File: 백준/Silver/1629. 곱셈/곱셈.py
A, B, C = [int(x) for x in input().split()]

remainder = A % C
more_multiple = []

while B != 1:
    if B % 2 != 0:
        more_multiple.append(remainder)
    remainder = remainder * remainder % C
    B = B // 2

while more_multiple:
    remainder = remainder * more_multiple.pop(-1) % C

print(remainder)

File: 백준/Silver/1629. 곱셈/곱셈.swift
let ABC = readLine()!.split(separator: " ").compactMap { Int($0) },
    a = ABC[0],
    c = ABC[2]
    
var b = ABC[1],
    remainder = a % c,
    remainderStack = [Int]()
    
while b != 1 {
    if !b.isMultiple(of: 2) { remainderStack.append(remainder) }
    remainder = remainder * remainder % c
    b = b / 2
}

while !remainderStack.isEmpty {
    remainder = remainder * remainderStack.removeLast() % c
}

print(remainder)